src/Symfony/Component/Mailer/Transport/Smtp/SmtpTransport.php

```diff
@@ -146,11 +147,33 @@ public function send(RawMessage $message, Envelope $envelope = null): ?SentMessa
             throw $e;
         }
 
+        $this->parseMessageId($message);
         $this->checkRestartThreshold();
 
         return $message;
     }
 
+    public function parseMessageId(SentMessage $message): void
```

Comment:
if the goal is to allow child classes to customize it, it should be protected, not public

Reply:
and if the child classes should customize it, it should receive `$this->mtaResult` as argument instead of accessing the private property inside it (which child classes cannot do), and I would rather return the message id than expect the method to mutate a provided SentMessage.